test/vinyl-luatest/select_consistency_test.lua

```diff
@@ -139,80 +198,63 @@ g.test_select_consistency = function(cg)
                     end
                     if not found then
                         ok = false
+                        err = string.format('consistency check failed: ' ..
+                                            'key %s from %s not found in %s',
+                                            t1[1], idx1, idx2)
                         break
                     end
                 end
                 if not ok then
-                    table.insert(failed, {
-                        iterator = iterator,
-                        idx1 = idx1,
-                        idx2 = idx2,
-                        res1 = res1,
-                        res2 = res2,
-                    })
-                    error('consistency check failed')
+                    log.error(err)
+                    error(err)
```

Comment:
May be it is worth to stop the test (setting `stop` to true) on read or write error.

Reply:
Test is limited in time by just 5 seconds. Optimizing the error case doesn't seem to be justified.